### PR TITLE
RUM mobile SDK docs restructure

### PR DESCRIPTION
**Pull request by rtrieu**
### What does this PR do? What is the motivation?

As per [DOCS-9231](https://datadoghq.atlassian.net/browse/DOCS-9231):

Restructures RUM Mobile docs so that they are organized by framework instead of by topic
### Merge instructions

Merge readiness:
- [ ] Ready for merge

Merge queue is enabled in this repo. To have it automatically merged after it receives the required reviews, create the PR (from a branch that follows the `<yourname>/description` naming convention) and then add the following PR comment:

```
/merge
```

### Additional notes

[DOCS-9231]: https://datadoghq.atlassian.net/browse/DOCS-9231?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: content/en/real_user_monitoring/mobile_and_tv_monitoring/roku/_index.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+## Overview
+
+Datadog Real User Monitoring (RUM) enables you to visualize and analyze the real-time performance and user journeys of your application's individual users.
+
+## Start monitoring Roku applications
+
+To get started with RUM for Roku, create an application and configure the RokuSDK.
+
+{{< whatsnext desc="This section includes the following topics:">}}
+  {{< nextlink href="/real_user_monitoring/mobile_and_tv_monitoring/roku/setup">}}<u>Setup</u>: Learn how to setup the Roku SDK, track background events, and send data when devices are offline.{{< /nextlink >}}
```
Comment (jhgilbert) [inline content/en/real_user_monitoring/mobile_and_tv_monitoring/roku/_index.md:26]:
```suggestion
  {{< nextlink href="/real_user_monitoring/mobile_and_tv_monitoring/roku/setup">}}<u>Setup</u>: Learn how to set up the Roku SDK, track background events, and send data when devices are offline.{{< /nextlink >}}
```

File: content/en/real_user_monitoring/mobile_and_tv_monitoring/android/_index.md [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+Datadog Real User Monitoring (RUM) enables you to visualize and analyze the real-time performance and user journeys of your application's individual users.
+
+The Datadog Android SDK supports Android 5.0+ (API level 21) and Android TV. It works for Android apps written using Java or Kotlin.
+
+## Start monitoring Android applications
+
+To get started with RUM for Android, create an application and configure the Android SDK.
+
+{{< whatsnext desc="This section includes the following topics:">}}
+  {{< nextlink href="/real_user_monitoring/mobile_and_tv_monitoring/android/setup">}}<u>Setup</u>: Learn how to setup the Android SDK, track background events, and send data when devices are offline.{{< /nextlink >}}
```
Comment (jhgilbert) [inline content/en/real_user_monitoring/mobile_and_tv_monitoring/android/_index.md:25]:
```suggestion
  {{< nextlink href="/real_user_monitoring/mobile_and_tv_monitoring/android/setup">}}<u>Setup</u>: Learn how to set up the Android SDK, track background events, and send data when devices are offline.{{< /nextlink >}}
```

File: content/en/real_user_monitoring/mobile_and_tv_monitoring/flutter/_index.md [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+## Overview
+
+Datadog Real User Monitoring (RUM) enables you to visualize and analyze the real-time performance and user journeys of your application's individual users.
+
+## Start monitoring Flutter applications
+
+To get started with RUM for Flutter, create an application and configure the Flutter SDK.
+
+{{< whatsnext desc="This section includes the following topics:">}}
+  {{< nextlink href="/real_user_monitoring/mobile_and_tv_monitoring/flutter/setup">}}<u>Setup</u>: Learn how to setup the Flutter SDK, track background events, and send data when devices are offline.{{< /nextlink >}}
```
Comment (jhgilbert) [inline content/en/real_user_monitoring/mobile_and_tv_monitoring/flutter/_index.md:23]:
```suggestion
  {{< nextlink href="/real_user_monitoring/mobile_and_tv_monitoring/flutter/setup">}}<u>Setup</u>: Learn how to set up the Flutter SDK, track background events, and send data when devices are offline.{{< /nextlink >}}
```

File: content/en/real_user_monitoring/mobile_and_tv_monitoring/ios/_index.md [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+## Overview
+
+Datadog Real User Monitoring (RUM) enables you to visualize and analyze the real-time performance and user journeys of your application's individual users.
+
+## Start monitoring iOS applications
+
+To get started with RUM for iOS, create an application and configure the iOS SDK.
+
+{{< whatsnext desc="This section includes the following topics:">}}
+  {{< nextlink href="/real_user_monitoring/mobile_and_tv_monitoring/ios/setup">}}<u>Setup</u>: Learn how to setup the iOS SDK, track background events, and send data when devices are offline.{{< /nextlink >}}
```
Comment (jhgilbert) [inline content/en/real_user_monitoring/mobile_and_tv_monitoring/ios/_index.md:24]:
```suggestion
  {{< nextlink href="/real_user_monitoring/mobile_and_tv_monitoring/ios/setup">}}<u>Setup</u>: Learn how to set up the iOS SDK, track background events, and send data when devices are offline.{{< /nextlink >}}
```

File: content/en/real_user_monitoring/mobile_and_tv_monitoring/kotlin_multiplatform/_index.md [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+Datadog Real User Monitoring (RUM) enables you to visualize and analyze the real-time performance and user journeys of your application's individual users.
+
+The Datadog Kotlin Multiplatform SDK supports Android 5.0+ (API level 21) and iOS v12+.
+
+## Start monitoring Kotlin Multiplatform applications
+
+To get started with RUM for Kotlin Multiplatform, create an application and configure the Kotlin Multiplatform SDK.
+
+{{< whatsnext desc="This section includes the following topics:">}}
+  {{< nextlink href="/real_user_monitoring/mobile_and_tv_monitoring/kotlin_multiplatform/setup">}}<u>Setup</u>: Learn how to setup the Kotlin Multiplatform SDK, track background events, and send data when devices are offline.{{< /nextlink >}}
```
Comment (jhgilbert) [inline content/en/real_user_monitoring/mobile_and_tv_monitoring/kotlin_multiplatform/_index.md:26]:
```suggestion
  {{< nextlink href="/real_user_monitoring/mobile_and_tv_monitoring/kotlin_multiplatform/setup">}}<u>Setup</u>: Learn how to set up the Kotlin Multiplatform SDK, track background events, and send data when devices are offline.{{< /nextlink >}}
```

File: content/en/real_user_monitoring/mobile_and_tv_monitoring/react_native/_index.md [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+## Overview
+
+Datadog Real User Monitoring (RUM) enables you to visualize and analyze the real-time performance and user journeys of your application's individual users.
+
+## Start monitoring React Native applications
+
+To get started with RUM for React Native, create an application and configure the React Native SDK.
+
+{{< whatsnext desc="This section includes the following topics:">}}
+  {{< nextlink href="/real_user_monitoring/mobile_and_tv_monitoring/react_native/setup">}}<u>Setup</u>: Learn how to setup the React Native SDK, track background events, and send data when devices are offline.{{< /nextlink >}}
```
Comment (jhgilbert) [inline content/en/real_user_monitoring/mobile_and_tv_monitoring/react_native/_index.md:26]:
```suggestion
  {{< nextlink href="/real_user_monitoring/mobile_and_tv_monitoring/react_native/setup">}}<u>Setup</u>: Learn how to set up the React Native SDK, track background events, and send data when devices are offline.{{< /nextlink >}}
```

File: content/en/real_user_monitoring/mobile_and_tv_monitoring/unity/data_collected.md [hunk 7]
```diff
@@ -1,11 +1,9 @@
 ---
-title: RUM Unity Data Collected
```
Comment (jhgilbert) [inline content/en/real_user_monitoring/mobile_and_tv_monitoring/unity/data_collected.md:2]: I think Data Collected still needs to be added under Unity in the nav, see [the staging page](https://docs-staging.datadoghq.com/rtrieu/rum-mobile-sdk-docs-restructure/real_user_monitoring/mobile_and_tv_monitoring/unity/data_collected/)

File: content/en/real_user_monitoring/mobile_and_tv_monitoring/unity/advanced_configuration.md [hunk 8]
```diff
@@ -1,15 +1,13 @@
 ---
-title: RUM Unity Advanced Configuration
+title: Unity Advanced Configuration
```
Comment (jhgilbert) [inline content/en/real_user_monitoring/mobile_and_tv_monitoring/unity/advanced_configuration.md:2]: I think Advanced Configuration still needs to be added under Unity in the nav, see [the staging page](https://docs-staging.datadoghq.com/rtrieu/rum-mobile-sdk-docs-restructure/real_user_monitoring/mobile_and_tv_monitoring/unity/advanced_configuration/)

**Review by jhgilbert — COMMENTED** [review]
Congrats for making all of this happen, this is a major reorg! 

In addition to the suggestions, I also found that two of the staging pages look a bit odd in that they don't have a corresponding nav entry. Were they supposed to be deleted / become aliases instead?

- [Web Tracking](https://docs-staging.datadoghq.com/rtrieu/rum-mobile-sdk-docs-restructure/real_user_monitoring/mobile_and_tv_monitoring/web_view_tracking/?tab=android)
- [Android Crash Reporting and Error Tracking](https://docs-staging.datadoghq.com/rtrieu/rum-mobile-sdk-docs-restructure/error_tracking/frontend/mobile/android/?tab=us)

Disregard if not, thank you!

File: content/en/real_user_monitoring/mobile_and_tv_monitoring/android/integrated_libraries.md [hunk 9]
```diff
@@ -1,15 +1,13 @@
...
-code_lang_weight: 10
 aliases:
 - /real_user_monitoring/android/integrated_libraries/
+- /real_user_monitoring/mobile_and_tv_monitoring/android/integrated_libraries/
 further_reading:
 - link: https://github.com/DataDog/dd-sdk-android
   tag: "Source Code"
   text: Source code for dd-sdk-android
-- link: /real_user_monitoring/mobile_and_tv_monitoring/integrated_libraries
+- link: /real_user_monitoring/mobile_and_tv_monitoring/android/integrated_libraries
```
[the remark this answers is not in the document]
Comment (rtrieu) [reply content/en/real_user_monitoring/mobile_and_tv_monitoring/android/integrated_libraries.md:10]: removed, thank you!

File: content/en/real_user_monitoring/mobile_and_tv_monitoring/unity/setup.md [hunk 10]
```diff
@@ -23,9 +21,9 @@ further_reading:
 ---
 ## Overview
 
-{{< callout url="#" btn_hidden="true" header="false">}}
-Unity Monitoring is in Preview.
-{{< /callout >}}
+{{< beta-callout url="#" btn_hidden="true" >}}
+Unity Monitoring is in public beta.
```
[the remark this answers is not in the document]
Comment (rtrieu) [reply content/en/real_user_monitoring/mobile_and_tv_monitoring/unity/setup.md:25]: This was not intentional :) reverting! Thanks for the catch.

**Comment by rtrieu** [comment]
Hi @jhgilbert, thank you for catching all the things you did in the last review, which I believe I've addressed! Can you give it another pass? Here are some significant new updates:
- Expo and CodePush are no longer part of the React Native multi-code-lang pages, and are now their own sections with reused content
- Added link to the specific tabs for platforms on the Mobile Vitals and Web View Tracking pages
- Mobile Vitals has been removed for Roku and Unity, as it's not relevant for those platforms
- Reconciled some missing info from the Unity setup doc

**Comment by jhgilbert** [comment]
@rtrieu Thanks for the bump! All of the pages listed under removed files in the preview links are 404s ... should they redirect?

**Comment by rtrieu** [comment]
@jhgilbert Good call! Just added redirects for those. Also, I had to take back my update to put CodePush and Expo their own sections! They are now back nested with React Native.

**Review by jhgilbert — APPROVED** [review]

File: layouts/partials/rum/rum-react-native-setup.html
Comment (StefonSimmons) [inline layouts/partials/rum/rum-react-native-setup.html:1]: @rtrieu I dont see where this partial is being used. where is it referenced

File: content/en/real_user_monitoring/platform/connect_rum_and_traces.md [hunk 11]
```diff
@@ -283,8 +283,8 @@ To start sending just your iOS application's traces to Datadog, see [iOS Trace C
     )..enableHttpTracking()
     ```
 
-[1]: /real_user_monitoring/mobile_and_tv_monitoring/setup/flutter/
-[2]: /real_user_monitoring/mobile_and_tv_monitoring/setup/flutter/#automatic-resource-tracking
+[1]: /real_user_monitoring/mobile_and_tv_monitoring/flutter/etup/
```
Comment (StefonSimmons) [inline content/en/real_user_monitoring/platform/connect_rum_and_traces.md:286]:
```suggestion
[1]: /real_user_monitoring/mobile_and_tv_monitoring/flutter/setup/
```

File: content/en/real_user_monitoring/mobile_and_tv_monitoring/kotlin_multiplatform/setup.md [hunk 12]
```diff
@@ -112,30 +110,27 @@ If you are integrating Kotlin Multiplatform library as a framework with an `embe
 {{< tabs >}}
 {{% tab "RUM" %}}
 
-1. Navigate to [**Digital Experience** > **Add an Application**][1].
+1. Navigate to [**Digital Experience** > **Add an Application**][2].
```
Comment (StefonSimmons) [inline content/en/real_user_monitoring/mobile_and_tv_monitoring/kotlin_multiplatform/setup.md:113]: @rtrieu does this have the correct link?

File: layouts/partials/rum/rum-react-native-setup.html
Comment (rtrieu) [reply layouts/partials/rum/rum-react-native-setup.html:1]: Ah, I need to delete that. Previously, I had it here: content/en/real_user_monitoring/mobile_and_tv_monitoring/react_native/setup/_index.md

File: content/en/real_user_monitoring/mobile_and_tv_monitoring/kotlin_multiplatform/setup.md (hunk 12 above)
Comment (rtrieu) [inline content/en/real_user_monitoring/mobile_and_tv_monitoring/kotlin_multiplatform/setup.md:113]:
```suggestion
1. Navigate to [**Digital Experience** > **Add an Application**][1].
```

Comment (rtrieu) [inline content/en/real_user_monitoring/mobile_and_tv_monitoring/kotlin_multiplatform/setup.md]:
```suggestion
1. Navigate to [**Digital Experience** > **Add an Application**][1].
```

Comment (rtrieu) [reply content/en/real_user_monitoring/mobile_and_tv_monitoring/kotlin_multiplatform/setup.md]: Updated to [1], thanks!

Comment (rtrieu) [inline content/en/real_user_monitoring/mobile_and_tv_monitoring/kotlin_multiplatform/setup.md]:
```suggestion
1. Navigate to [**Digital Experience** > **Add an Application**][1].
```

**Review by StefonSimmons — APPROVED** [review]

**Pull request merged**